sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -490,19 +493,38 @@ private HttpPipeline setupPipeline(HttpClient httpClient) {
                 throw logger.logExceptionAsError(Exceptions.propagate(e));
             }
         }).map(ar -> new MsalToken(ar, options))
-        .filter(t -> !t.isExpired())
-        .switchIfEmpty(Mono.fromFuture(() -> {
-            SilentParameters.SilentParametersBuilder forceParametersBuilder = SilentParameters.builder(
-                new HashSet<>(request.getScopes())).forceRefresh(true);
-            if (account != null) {
-                forceParametersBuilder = forceParametersBuilder.account(account);
-            }
+            .filter(t -> !t.isExpired())
+            .switchIfEmpty(Mono.fromFuture(() -> {
+                SilentParameters.SilentParametersBuilder forceParametersBuilder = SilentParameters.builder(
+                    new HashSet<>(request.getScopes())).forceRefresh(true);
+                if (account != null) {
+                    forceParametersBuilder = forceParametersBuilder.account(account);
+                }
+                try {
+                    return getPublicClientApplication(false).acquireTokenSilently(forceParametersBuilder.build());
+                } catch (MalformedURLException e) {
+                    throw logger.logExceptionAsError(Exceptions.propagate(e));
```

Comment:
Does `Mono.fromFuture` wrap the thrown exception into a `Mono.error` or will this need to be handled as a regular exception?

Reply:
I'm not sure how reactor handles it. Since it's not documented even if there's anything, I'm returning CompletableFuture.failedFuture() instead.